### PR TITLE
Changes in error cases

### PR DESCRIPTION
**Pull request by apoorva11029**

### CODE REVIEW

File: pkg/types/openstack/validation/platform.go [hunk 1]
```diff
@@ -85,11 +85,36 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
...
                         }
                 }
         }
+
+	machineMask := n.MachineCIDR.Mask
+	if p.AciNetExt.NeutronCIDR.String() != "" {
+                neutronMask := p.AciNetExt.NeutronCIDR.Mask
+                if machineMask.String() != neutronMask.String() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("NeutronCIDR"),
+					p.AciNetExt.NeutronCIDR.String(), "Machine CIDR and Neutron CIDR have different subnet masks"))
```
Comment (snaiksat) [inline pkg/types/openstack/validation/platform.go:98]: nit: Here and elsewhere below, let's have a uniform convention when we refer to configuration options that we want to indicate are incorrectly specified. I would suggest that we refer to the exact configuration key, so in the above case the I suggest the message:
"The CIDRs specified in the machineCIDR (%s) and neutronCIDR (%s) configurations in the install-config.yaml have different subnet masks (machineCIDR mask: %s, neutronCIDR mask: %s)"

Comment (apoorva11029) [reply pkg/types/openstack/validation/platform.go]: Done

File: pkg/types/openstack/validation/platform.go [hunk 2]
```diff
@@ -85,11 +85,35 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
...
+                                                Mask: machineMask,
+                                        },
+                                }
+        }
+        n.NeutronCIDR = p.AciNetExt.NeutronCIDR
+
+	_, err = ipnet.ParseCIDR(p.AciNetExt.InstallerHostSubnet)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("InstallerHostSubnet"),
+                		p.AciNetExt.InstallerHostSubnet, "InstallerHostSubnet has an invalid subnet value (" + p.AciNetExt.InstallerHostSubnet + ")"))
```
Comment (snaiksat) [inline pkg/types/openstack/validation/platform.go:114]: Hate to nit pick again, but can you  make the config names exactly the same way they are in the config file, meaning, starting with a lower case, i.e. installerHostSubnet
The idea is that if someone copies the config name from the above message, and tries to search in the install-config, they should be able to find it. In my opinion, this point takes precedence over adhering to grammar rules in such cases. And also please follow that pattern consistently elsewhere as well.

Comment (apoorva11029) [reply pkg/types/openstack/validation/platform.go:114]: It's not possible to have config name start with lowercase here (golang convention for JSON structs)

**Pull request merged**